datafusion/sqllogictest/test_files/join.slt

```diff
@@ -614,8 +614,8 @@ query TI
 SELECT t1.a, t1.b FROM t1 JOIN t2 ON t1.a = t2.a ORDER BY t1.a, t2.b;
 ----
 Alice 50
-Alice 50
 Alice 100
+Alice 50
```

Comment:
Seems that after this PR the results matches expected ones, in comment above.
Should we update the comment? 

> the current query results **are incorrect**, becuase the query was incorrectly rewritten as:

Reply:
Probably we should have both test-cases (just in case):

- `SELECT t1.a, t1.b FROM t1 JOIN t2 ON t1.a = t2.a ORDER BY t1.a, t1.b;`
```
Alice 	50
Alice 	50
Alice 	100
Alice 	100
```

- `SELECT t1.a, t1.b FROM t1 JOIN t2 ON t1.a = t2.a ORDER BY t1.a, t2.b;`
```
Alice 	50
Alice 	100
Alice 	50
Alice 	100
```

Reply:
Nice catch. Yea, this existing test now matches correct behaviors after this PR. Actually it is same issue (order by with same column name in joined sides).

Updated the test case.

Reply:
> It seems it would be useful to have a set of tests that can fail, but do not block CI, etc (something like TDD, we know it fails and we can make it work)

I agree this would be useful -- we can do it in the rust tests with `#[ignore]`. Maybe we need to add something similar to sqllogictests


> It will be much easier to find test-cases that don't work right now. The tests won't even have to be changed, except perhaps excluded from the "expectedly fail" category.

One way to do this is to search for issue links `https://github.com...`